### ISSUE TITLE
attachement exploit (without backdoor)

### ISSUE THREAD

**Issue by mrrobotnl** [rapid7/metasploit-framework#15718]
## Summary
I am currently working on a project that include evil attachments. The main goal of this project is to look who has pressed on the attachment. behind the attachment there will be a script that collects some basic identifier and sends it back to be full circle. But I am not sure what exploit I can use for this. 

The attachment can be in the following formats: 
- Pdf 
- Image 
- or office product (like word excel or power point) 

And I can write the script in either Powershell, python (as exe) or in VBA. 

I already tried a couple of adobe pdf exploits but never seems to get it working. I am looking for an exploit without a backdoor, this is out of the scope that I intent to work in. 

In conclusion 
Are there any tips or suggestion would be awesome. Thanks in advance.

**Comment by timwr** [comment]
I believe pingback payloads were made for this purpose:
https://www.rapid7.com/blog/post/2019/08/01/introducing-pingback-payloads/

**Comment by mrrobotnl** [comment]
> 
> 
> I believe pingback payloads were made for this purpose: https://www.rapid7.com/blog/post/2019/08/01/introducing-pingback-payloads/

Thanks for the fast reaction, I will look into it. If I am understand it correctly, I could use this behind a pdf file right?

**Comment by bcoles** [comment]
> Thanks for the fast reaction, I will look into it. If I am understand it correctly, I could use this behind a pdf file right?

You can, if the file is opened in an outdated and vulnerable PDF reader.

You're looking for something more like a canary token.

* https://docs.canarytokens.org/guide/adobe-pdf-token.html
* https://docs.canarytokens.org/guide/ms-word-token.html

**Comment by bcoles** [comment]
As far as I'm aware there's no functionality for canary tokens / pingbacks in Metasploit that does not require code execution.

> I am looking for an exploit without a backdoor

By backdoor I presume you're referring to a bind shell / reverse shell (a "session" in Metasploit vernacular).

Some privacy-violating features could be "exploited" for this purpose (ie, to load remotely-hosted resources from within a spreadsheet, word doc, etc), but generally speaking, Metasploit is used for exploitation leading to code execution.

While the "pingback" payloads somewhat match your use case (in that they are non-malicious callbacks) these still require code execution on the host to connect back.

> I already tried a couple of adobe pdf exploits

For your use case, using code execution is significantly less effective than abusing intended functionality.

Code execution is also significantly more error prone. For example, a MS Word exploit will be useless if the file is opened in Libre Office; an Adobe Reader exploit is useless if the file is opened in Nitro PDF, etc.

Worse, even if the target opens the file in the correct program, the program may have been patched, or had additional workaround patches applied, or configuration mitigations in place, and end point protection may prevent opening the file, and border proxies/firewalls may detect and block the callback, and ...

> The main goal of this project is to look who has pressed on the attachment. behind the attachment there will be a script that collects some basic identifier and sends it back to be full circle.

Unfortunately, gathering host info will likely require executing code on the host. It is possible that you could harvest some host information from HTTP request headers when using a document which loads remote resources from a web server; however, the details will not be sufficient for user identification.

If you're the network administrator for the network you could leverage whatever ~~spyware~~ employee monitoring software you're using to check which files have been opened. This approach is also not foolproof (file can be renamed, BYOD?, file can be copied to another host, etc).

Using a HTML file which loads an image from a remote web server (with a unique identifier as the file name) may be a simple universal option (works on all operating systems, all browsers, etc), but users may be more wary of HTML files than other formats.

Instead, I suggest using canary tokens (as per https://github.com/rapid7/metasploit-framework/issues/15718#issuecomment-929099707) and ensuring each generated file contains a unique token per document per recipient.

Closing this issue.
